### ISSUE TITLE
Bob Storage: Upload or Download files .NET sample using Shared Access Signature (SAS)

### ISSUE THREAD

**Issue by BrianHo888** [Azure/azure-sdk-for-net#19425]
Hi,
Is there a C# client application sample you can share to walk through how to upload or download file(s) using SAS?  I have seen different samples of how to create and authenticate SAS for clients to consume but did not file any sample on the client side how to use the SAS Uri to upload or download files.  It would be nice to include some samples.

Thanks,
Brian.

**Comment by jsquire** [comment]
Hi @BrianHo888.  Thank you for your feedback and we regret that you're experiencing difficulties.   I'm not able to determine from the context of your issue which SDK you're using.  Would you be so kind as to provide a bit more detail so that I can engage the right folks to assist?

**Comment by BrianHo888** [comment]
> Hi,
> Is there a C# client application sample you can share to walk through how to upload or download file(s) using SAS? I have seen different samples of how to create and authenticate SAS for clients to consume but did not file any sample on the client side how to use the SAS Uri to upload or download files. It would be nice to include some samples.
> 
> Thanks,
> Brian.



> Hi @BrianHo888. Thank you for your feedback and we regret that you're experiencing difficulties. I'm not able to determine from the context of your issue which SDK you're using. Would you be so kind as to provide a bit more detail so that I can engage the right folks to assist?

I follow https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-dotnet to develop a console app to automate the upload and download files from Azure Blob.  It is working great with the sample provided.  Above link provide the sample of ConnectionString but I can't find any sample to use Shared Access Signature (SAS) to upload and download from the client side.  I found some samples from docs.microsoft.com to create container and SAS authentication.  In most cases, vendor would provide the SAS Uri for us to send and retrieve files. 

We have been given SAS Uri to retrieve data from vendor's Blob.  For security reason, they don't provide AccountKey or ConnectionString to access their Azure Blob storage.  Only SAS Uri is provided.  

I am looking for C# .NET code to send and retrieve files via SAS Uri connection.

Thanks,
Brian.

**Comment by jsquire** [comment]
Thank you for the additional context, @BrianHo888.  Adding some tags and routing this over to the team best able to assist.

**Comment by seanmcc-msft** [comment]
Hi @BrianHo888,

Here is our SAS sample - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/samples/Sample02_Auth.cs#L137

And here is our download sample - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/samples/Sample01b_HelloWorldAsync.cs#L66

If you have an existing SAS URL, you can download the blob like this:

```
BlobClient blobClient = new BlobClient(new Uri("SAS URL"));
Response<BlobDownloadInfo> response = await blobClient .DownloadAsync();
```

-Sean

**Comment by tafs7** [comment]
@seanmcc-msft 
hate to comment on a closed issue, but noticed in v12.9.0 that the `DownloadAsync` method is no longer browsable in VS Intellisense (the decompiled code show it marked with EditorBrowsable.Never).

Is that API going away, and should we be downloading blobs using a different API, now? (I see `DownloadStreamingAsync`, `DownloadContentAsync`, and `DownloadToAsync`).

All the samples that I see still use `DownloadAsync()`.
